sympy/functions/special/delta_functions.py

```diff
@@ -18,6 +18,17 @@ class DiracDelta(Function):
     """
     The DiracDelta function and its derivatives.
 
+    DiracDelta is not an ordinary function. It can be rigorously defined either
+    as a distribution or as a measure.
+
+    As a measure, it accepts a subset A of the real number line as an argument,
+    and returns DiracDelta(A) = 1 if 0 belongs to A, and DiracDelta(A) = 0 otherwise. Only
+    these integrated values are well defined, not the pointwise values of the measure itself.
+
+    In the theory of distributions, DiracDelta is defined only in relation to how it affects
+    other functions when it is "integrated" against them. It simply takes a good test function ``f(x)``
+    to its value ``f(0)`` whereas the derivative of DiracDelta takes a test function ``f(x)`` to ``-f'(0)``.
+
```

Comment:
We need a sentence or two here that explains what this all means in terms of use in SymPy. Some questions:

1. Does DiracDelta behave like other functions in SymPy?
2. Are there any odd things that make this stand out wrt to other functions?
3. What is the expected behavior of DiracDelta?
etc